src/pages/storage/StorageVolumesFilter.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { FC, useEffect } from "react";
+import { SearchAndFilter } from "@canonical/react-components";
+import {
+  SearchAndFilterData,
+  SearchAndFilterChip,
+} from "@canonical/react-components/dist/components/SearchAndFilter/types";
+import { useLocation } from "react-router-dom";
+import { LxdStorageVolume } from "types/storage";
+
+interface StorageFilterState {
+  state?: {
+    appliedPool: string;
+  };
+}
+
+export interface StorageVolumesFilterType {
+  queries: string[];
+  pools: string[];
+  volumeTypes: string[];
+  contentTypes: string[];
+}
+
+interface Props {
+  volumes: LxdStorageVolume[];
+  setFilters: (newFilters: StorageVolumesFilterType) => void;
+}
+
+const volumeTypes: string[] = [
+  "container",
+  "virtual-machine",
+  "snapshot",
+  "image",
+  "custom",
+];
+
+const contentTypes: string[] = ["block", "filesystem", "iso"];
+
+const POOL = "Pool";
+const VOLUME_TYPE = "Volume type";
+const CONTENT_TYPE = "Content type";
+
+const StorageVolumesFilter: FC<Props> = ({ volumes, setFilters }) => {
+  const { state } = useLocation() as StorageFilterState;
+
+  useEffect(() => window.history.replaceState({}, ""), [state]);
+
+  const pools = [...new Set(volumes.flatMap((volume) => volume.pool))];
+
+  const searchAndFilterData: SearchAndFilterData[] = [
+    {
+      id: 1,
+      heading: POOL,
+      chips: pools.map((pool) => {
+        return { lead: POOL, value: pool };
+      }),
+    },
+    {
+      id: 2,
+      heading: VOLUME_TYPE,
+      chips: volumeTypes.map((status) => {
+        return { lead: VOLUME_TYPE, value: status };
+      }),
+    },
+    {
+      id: 3,
+      heading: CONTENT_TYPE,
+      chips: contentTypes.map((type) => {
+        return { lead: CONTENT_TYPE, value: type };
```

Comment:
Maybe `contentType` is a bit more informative?
```suggestion
      chips: contentTypes.map((contentType) => {
        return { lead: CONTENT_TYPE, value: contentType };
```